backend/src/services/UserService.ts

```diff
@@ -1,53 +1,129 @@
-
+/**
+ * -----------------------------------------------------------------------------
+ * Imports
+ * -----------------------------------------------------------------------------
+ */
+import { resolve } from 'path';
 import { Service, Inject } from 'typedi';
 import UserProfileDAO from '../integrations/UserProfileDAO';
 import { IUserProfileInput, IUserProfile } from '../interfaces/IUser';
 import ScaleService from './ScaleService';
 const { v4: uuidv4 } = require('uuid');
 
+/**
+ * -----------------------------------------------------------------------------
+ * Service class
+ * -----------------------------------------------------------------------------
+ */
 @Service()
 class UserService {
+  /**
+   * ---------------------------------------------------------------------------
+   * Attributes
+   * ---------------------------------------------------------------------------
+   */
   userDao: UserProfileDAO;
   @Inject()
   scaleService: ScaleService;
 
-  constructor() {
+  /**
+   * ---------------------------------------------------------------------------
+   * Method
+   * ---------------------------------------------------------------------------
+   */
+   public constructor() {
     this.userDao = new UserProfileDAO;
   }
 
+  /**
+   * ---------------------------------------------------------------------------
+   * Method
+   * ---------------------------------------------------------------------------
+   */
   private generateConsentCode = () => Math.floor(1000 + Math.random() * 9000);
 
-  addNewUser(userInput:IUserProfileInput):Promise<void> {
+  /**
+   * ---------------------------------------------------------------------------
+   * Method
+   * ---------------------------------------------------------------------------
+   */
+  public addNewUser(userInput:IUserProfileInput):Promise<void> {
     var user: IUserProfile = {
       id: uuidv4(),
       ...userInput,
       consentCode: this.generateConsentCode()
     }
 
     this.userDao.addUser(user);
-
     return this.scaleService.addUserToScale(user).then(
       userProfile => {
         console.log(`User added to scale with index: ${userProfile.index}`);
-        this.userDao.addUser(user);
-        return new Promise((resolve) => resolve());
+        this.userDao.updateUser(userProfile);
+        return Promise.resolve();
       }
-    )
+    ).catch(() => {
+      this.userDao.deleteUserById(user.id);
```

Comment:
instead of adding and deleting like this. Let's just do an add where you do updateUser now. If addUserToScale fails you won't save user to userDao.

Reply:
not possible, the user must exist during the registration on the scale, otherwise the required measurement will not be associated with a user and the current backend implementation will crash.